### PR TITLE
Follow-ups on detailed confirmation page, part 1

### PR DESCRIPTION
**Pull request by kalvinwang**
This PR:

- Makes the entire gray bar on the detailed confirmation page clickable, and refactors the gray bar into a common component ("AccordionItem") 
- Adds missing snapshots
- Restores missing content changes and aria-label on the gray bars that were lost in a previous merge (so it now actually looks like the screenshots from #487)

===

Partially completes #489

- [x] Snapshots updated, if output HTML/JS has changed (`npm run test:update-snapshots`)
- [ ] Changes reviewed on mobile using devtools, if output HTML/CSS has changed
- [ ] Spanish checked by adding `?lng=es` to URL e.g. `/guide/benefits?lng=es`, if Spanish updated
- [ ] Automated tests added, if new functionality added
- [ ] Documentation (e.g. READMEs) updated, if relevant
- [ ] IE11 and Edge compatibility confirmed via manual testing in Browserstack, if new libraries added or newish JS/HTML/CSS features used for the first time in this codebase
- [ ] Accessibility & performance audit run using Google Lighthouse, if major changes made

### CODE REVIEW

File: src/client/components/AccordionItem/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        </div>
+      </Alert>
+      {showDetail && <div className="detail">{expandedBody}</div>}
+    </React.Fragment>
+  );
+}
+
+AccordionItem.propTypes = {
+  header: PropTypes.element,
+  expandedBody: PropTypes.element,
```
Comment (idealisms) [inline src/client/components/AccordionItem/index.js:34]: Maybe mark these as isRequired?

File: src/client/components/ListOfWeeksWithDetail/index.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      refuseWork: false,
+      schoolOrTraining: false,
+      workOrEarn: false,
+    };
+
+    const wrapper = renderNonTransContent(Component, "ListOfWeeksWithDetail", {
+      userData: {
+        formData: weekOfAnswers,
+        weeksToCertify: [1, 2, 3],
+        programPlan: "UI full time",
```
Comment (idealisms) [inline src/client/components/ListOfWeeksWithDetail/index.test.js:19]: Hmm, does this need to be an array? The snapshot has weekProgramPlan equal to individual letters.

File: src/client/components/AccordionItem/index.js (hunk 1 above)
Comment (kalvinwang) [reply src/client/components/AccordionItem/index.js:34]: Oh good point, I should do that for the other components I've created too. Thanks!

File: src/client/components/ListOfWeeksWithDetail/index.test.js (hunk 2 above)
Comment (kalvinwang) [reply src/client/components/ListOfWeeksWithDetail/index.test.js:19]: good catch! funny, proptypes validates that it's an array of strings, but I guess in javascript a string qualifies as an array of strings? :p

**Pull request merged**